addons/mail/static/src/chatter/web/form_renderer.js

```diff
@@ -43,4 +44,22 @@ patch(FormRenderer.prototype, {
         });
         return this.messagingState.thread.attachmentsInWebClientView.length > 0;
     },
+    mailLayout(hasAttachmentContainer) {
+        const xxl = this.uiService.size >= SIZES.XXL;
+        const hasFile = this.hasFile();
+        const hasChatter = !!this.mailStore;
+        const hasExternalWindow = !!this.externalWindowService.externalWindow;
+        if (hasExternalWindow && hasFile && hasAttachmentContainer) {
+            if (xxl) return "external_combo_xxl";  //  chatter on the side, attachment in separate tab
+            return "external_combo"  // chatter on the bottom, attachment in separate tab
+        }
+        if (hasChatter) {
+            if (xxl) {
+                if (hasAttachmentContainer && hasFile) return "combo";  // chatter on the bottom, attachment on the side
+                return "side_chatter";  // chatter on the side, no attachment
+            }
+            return "bottom_chatter";  // chatter on the bottom, no attachment
+        }
+        return "none";
+    }
```

Comment:
Would be easier to understand all states with enum, so something like:
```js
MAIL_LAYOUT = {
   NONE: 0,
   SIDE: 1,
   COMBO: 2,
   COMBO_EXTERNAL: 3,
   COMBO_EXTERNAL_XXL: 4,
}
```
Advantages of this is that it's well documented, the ordering makes sense, and in template these conditions on state are easy to spot thanks to `__comp__.mailLayout == __comp__.MAIL_LAYOUT.COMBO`, reducing risks to make typos with hard-coded strings.

Reply:
Sure, you can still make a typo with `MAIL_LAYOUT.NONNE` though for instance :wink:

Reply:
I didn't end up doing this one.
I tried it, but it ends up making the code harder to read with no benefits because we can still make typos (and the static code checker doesn't help since we are writing code in strings)

Reply:
This is less likely in JS thanks to autocomplete. In XML this sucks, but at least the UPPERCASE are easier to spot than the hard-coded strings. This change looks irrelevant but it actually helps with working with the different states. :)

Reply:
Can I go for uppercase strings then?

Reply:
Ok, this is a good compromise :)